src/backend/base/langflow/components/tools/currency_convertor.py

```
@@ -0,0 +1,130 @@
+        resp.raise_for_status()
+        payload = resp.json()
+        if not payload.get("success", False):
+            raise Exception(f"API error: {payload.get('error')}")
+
+        timestamp = datetime.utcfromtimestamp(payload.get("timestamp")).isoformat() + "Z"
+        quotes = payload.get("quotes", {})
+
+        try:
+            amount_val = float(self.amount)
+        except (TypeError, ValueError):
+            raise Exception("❌ ‘amount’ must be a number (e.g. 12 or 12.34).")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)